pkgpanda/http/__init__.py

```diff
@@ -0,0 +1,170 @@
+"""Pkgpanda HTTP API"""
+
+import http.client
+import logging
+import sys
+
+from flask import Flask, current_app, jsonify, request
+
+from pkgpanda import Install, Repository, actions
+from pkgpanda.exceptions import PackageError, ValidationError
+
+
+empty_response = ('', http.client.NO_CONTENT)
+
+
+def package_listing_response(package_ids):
+    return jsonify(sorted(package_ids))
+
+
+def package_response(package_id, repository):
+    try:
+        package = repository.load(package_id)
+    except PackageError:
+        error_message = 'Unable to load package {}.'.format(package_id)
+        logging.exception(error_message)
+        return error_response(error_message), http.client.INTERNAL_SERVER_ERROR
+
+    return jsonify({
+        'id': str(package.id),
+        'name': str(package.name),
+        'version': str(package.version),
+    })
+
+
+def error_response(message):
+    return jsonify({'error': message})
+
+
+app = Flask(__name__)
+app.config.from_object('pkgpanda.http.config')
+app.config.from_envvar('PKGPANDA_HTTP_CONFIG', silent=True)
+
+
+@app.errorhandler(Exception)
+def uncaught_exception_handler(exc):
+    error_message = 'An unexpected error has occurred.'
+    logging.error(error_message, exc_info=exc)
+    return error_response(error_message), http.client.INTERNAL_SERVER_ERROR
+
+
+@app.before_request
+def set_app_attrs_from_config():
+    current_app.install = Install(
+        current_app.config['DCOS_ROOT'],
+        current_app.config['DCOS_CONFIG_DIR'],
+        current_app.config['DCOS_ROOTED_SYSTEMD'],
+        manage_systemd=True,
+        block_systemd=False)
+    current_app.repository = Repository(
+        current_app.config['DCOS_REPO_DIR'])
+
+
+@app.route('/', methods=['GET'])
+def get_package_list():
+    return package_listing_response(current_app.repository.list())
+
+
+@app.route('/<package_id>', methods=['GET'])
```

Comment:
Can we have this and the POST / fetch_package be `/packages/<package_id>` rather than this?

As this stands, it makes it hard to know whether a request is going to go into this route, or things like `/active`. Effectively we've merged those key spaces and made it possible for people to do something ambiguous. I'd much rather keep them separate.